crawl-ref/source/travel.cc

```diff
@@ -4750,8 +4750,15 @@ bool check_for_interesting_features()
         if (env.map_knowledge(p).flags & MAP_SEEN_FLAG
             && !env.map_seen(p))
         {
+            // Update the shadow map
             env.map_seen.set(p);
-            _check_interesting_square(p, discoveries);
+
+            // But don't stop if we knew about it previously
+            if (env.map_forgotten.get()
+                && !((*env.map_forgotten.get())(p).flags & MAP_SEEN_FLAG))
```

Comment:
unique_ptr implements `operator bool` and `operator*`, so the `.get()` calls here should be superfluous

Reply:
Thanks for the tip. There are superfluous get()s elsewhere in the crawl code.